editor/components/block-settings-menu/block-settings-menu-plugins-item.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { difference } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { IconButton } from '@wordpress/components';
+import { compose } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import BlockSettingsMenuPluginsGroup from './block-settings-menu-plugins-group';
+
+const shouldRenderItem = ( selectedBlockNames, allowedBlockNames ) => ! Array.isArray( allowedBlockNames ) ||
+	difference( selectedBlockNames, allowedBlockNames ).length === 0;
+
+const BlockSettingsMenuPluginsItem = ( { allowedBlocks, icon, label, onClick, small, role } ) => (
+	<BlockSettingsMenuPluginsGroup>
+		{ ( { selectedBlocks, onClose } ) => {
+			if ( ! shouldRenderItem( selectedBlocks, allowedBlocks ) ) {
+				return null;
+			}
+			return ( <IconButton
+				className="editor-block-settings-menu__control"
+				onClick={ compose( onClick, onClose ) }
```

Comment:
compose will call: onClick( onClose( ...arg ) ); I think what we want here is:
```
( ...args ) => {
	onClick( ...args );
	onClose();
}
```

Reply:
Note that the `onClick` API isn't passed any props, so 

`onClick( onClose() );`

VS

```
( ) => {
	onClick( );
	onClose();
}
```
is essentially the same (omiting the sequence they're called, which doesn't matter as closing and executing the action are parallel processes).

Reply:
I thought we passed arguments to onClick, if we don't pass them it is exactly the same.